### PR TITLE
TransformControls: Derive from `Controls`.

### PR DESCRIPTION
**Pull request by Mugen87**
Related issue: https://github.com/mrdoob/three.js/pull/29085

**Description**

Same as https://github.com/mrdoob/three.js/pull/29085.

This is a breaking change though. `TransformControls` was derived from `Object3D` so far which had pros and cons. You could add the controls directly to the scene to render the gizmo (the visual helpers) but the `Object3D` methods `attach()` and `detach()` were overwritten with control specific methods.

The new approach is conceptually cleaner and more consistent compared to all other controls. But since `TransformControls` is no 3D object anymore, developers must add the gizmo differently now. So instead of:
```js
scene.add( controls );
```
It is now:
```js
scene.add( controls.getGizmo() );
```

### CODE REVIEW

File: examples/misc_controls_transform.html [hunk 1]
```diff
@@ -89,7 +88,9 @@
 				scene.add( mesh );
 
 				control.attach( mesh );
-				scene.add( control );
+
+				const gizmo = control.getGizmo();
```
Comment (Mugen87) [inline examples/misc_controls_transform.html:92]: @mrdoob Are you okay with `getGizmo()`?

**Review by puxiao — COMMENTED** [review]
TransformControls.html

You forgot to delete these: 

```diff
- <h3>[property:Object3D object]</h3>
- <p>
-	The 3D object being controlled.
- </p>
```

**Comment by Mugen87** [comment]
@puxiao Actually I had to restore some deleted docs. The `camera` property actually represents the camera whereas the `object` reference is overwritten with the object that is controlled by `TransformControls`.

**Comment by puxiao** [comment]
Rome wasn't built in a day.

File: examples/misc_controls_transform.html (hunk 1 above)
Comment (Mugen87) [reply examples/misc_controls_transform.html:92]: @arodic How does this change look to you?

**Comment by puxiao** [comment]
@Mugen87  Will these PRs be released in r168?
I have prepared some Chinese translations, but have not submitted PR yet, waiting for your this PR.

**Comment by Mugen87** [comment]
It has been a month since the PR was filed so I'm going ahead and merge it so we can finish the migration to `THREE.Controls`. 

Of course we can update the API at any point if things like `getGizmo()` or the connect/disconnect routines require changes.

File: examples/misc_controls_transform.html (hunk 1 above)
Comment (mrdoob) [reply examples/misc_controls_transform.html:92]: `getGizmo()` is okay 👌

Comment (mrdoob) [reply examples/misc_controls_transform.html:92]: Actually, `getHelper()` would follow the current naming: 
https://github.com/mrdoob/three.js/tree/dev/examples/jsm/helpers

**Comment by VegarRingdalAibel** [comment]
Updated from 167 to 169.
Got error when calling dispose(), error looks to be this:
https://github.com/mrdoob/three.js/blob/dev/examples/jsm/controls/TransformControls.js#L539-L544

Traverse was part of `Object3D` so dispose might need some changes. Unless Ive manged to overlook/mess something up.

**Comment by jacksontan** [comment]
Hi, how do you get the function getGizmo from controls? I cannot seem to see it in 
https://threejs.org/docs/#examples/en/controls/TransformControls or https://threejs.org/docs/#api/en/extras/Controls

As the suggested comment, this code `scene.add(controls)` will break on r169 and suggestion is to use `scene.add( controls.getGizmo() );` but cannot get the gizmo.

**Comment by VegarRingdalAibel** [comment]
image

**Pull request merged**